app/workers/import_worker.rb

```diff
@@ -0,0 +1,15 @@
+class ImportWorker
+  include Sidekiq::Worker
+
+  def perform(device, json_file)
```

Comment:
Okay. Other redical improvement could be switching to plain sql. Since we do create or update here, this will require of us to generate two separate queries: one for insert, one for update. But we can, probably, avoid it.